src/elements/content-explorer/ContentExplorer.js

```diff
@@ -1105,6 +1109,46 @@ class ContentExplorer extends Component<Props, State> {
         this.select(item, this.shareCallback);
     };
 
+    /**
+     * Fetch the shared link info
+     * @param {BoxItem} item - The item (folder, file, weblink)
+     * @returns {void}
+     */
+    fetchSharedLinkInfo = (item: BoxItem): void => {
+        const { id, type }: BoxItem = item;
+
+        switch (type) {
+            case TYPE_FOLDER:
+                this.api.getFolderAPI().getFolderFields(id, this.handleSharedLinkSuccess, noop, {
+                    fields: FILE_SHARED_LINK_FIELDS_TO_FETCH,
+                });
+                break;
+            case TYPE_FILE:
+                this.api
+                    .getFileAPI()
+                    .getFile(id, this.handleSharedLinkSuccess, noop, { fields: FILE_SHARED_LINK_FIELDS_TO_FETCH });
+                break;
+            case TYPE_WEBLINK:
+                break;
+            default:
+                throw new Error('Unknown Type');
+        }
+    };
+
+    /**
+     * Handles the shared link info by either creating a share link using enterprise defaults if
+     * it does not already exist, otherwise update the item in the state currentCollection.
+     *
+     * @param {Object} item file or folder
+     * @returns {void}
+     */
+    handleSharedLinkSuccess = (newItem: BoxItem) => {
+        const { currentCollection } = this.state;
+
+        // Update item in collection
+        this.updateCollection(currentCollection, newItem, () => this.setState({ isShareModalOpen: true }));
```

Comment:
Any way we can avoid calling `setState` twice?

Reply:
Not sure... the way to avoid it would be to have `updateCollection` not set state and defer to the caller but then that introduces more duplication throughout the code.